ttnn/cpp/ttnn/operations/ccl/reduce_scatter/device/reduce_scatter_op.cpp

```diff
@@ -88,8 +91,8 @@ Tensor reduce_scatter(
             const std::vector<Tensor>& input_tensors,
             const std::vector<std::optional<const Tensor>>& optional_input_tensors,
             const std::vector<std::optional<Tensor>>& optional_output_tensors) mutable -> std::vector<Tensor> {
-
-            bool is_ring = topology ==ttnn::ccl::Topology::Ring;
+            TT_ASSERT(input_tensors.size() >= 1, "Reduce scatter op expects an input tensor");
```

Comment:
In most cases we should use Fatal (not a fatal really, throws an exception) instead of assert (only checks in debug build)

Reply:
updated